File: BHS.Contracts/Blog/Post.cs
using System;

namespace BHS.Contracts.Blog
{
    public record Post(
        string Slug,
        string Title,
        string ContentMarkdown,
        Uri? FilePath,
        int? PhotosAlbumId,
        int? AuthorId,
        DateTimeOffset DatePublished,
        DateTimeOffset DateLastModified);
}

File: BHS.Web/ClientApp/src/app/data/schema/post.ts
import { Category } from './category';

export interface Post {
  slug: string;
  title: string;
  contentMarkdown: string;
  filePath: string;
  photosAlbumId?: number;
  authorId?: number;
  datePublished: Date;
  dateLastModified: Date;
  categories: Category[];
}
